rfcs/0185-redistribute-redistributable.md

```diff
@@ -0,0 +1,225 @@
+
+Hydra will build all packages with licenses for which `redistributable && runnableOnHydra`.
+It will still fail evaluation if the ISO image build or the Amazon AMIs were to contain any unfree software.
+
+This will be done by evaluating Nixpkgs twice in `release.nix`.
+Once with `allowUnfree = false` like today, plus once with `allowlistedLicenses = builtins.filter (l: l.redistributable && l.runnableOnHydra) lib.licenses`.
+Then, most of the jobs will be taken from the allowlisted nixpkgs, while only the builds destined for installation will be taken from the no-unfree nixpkgs.
+
+The list of jobs destined for installation, that cannot contain unfree software is:
+- `amazonImage`
+- `amazonImageAutomaticSize`
+- `amazonImageZfs`
+- `iso_gnome`
+- `iso_minimal`
+- `iso_minimal_new_kernel`
+- `iso_minimal_new_kernel_no_zfs`
+- `iso_plasma5`
+- `iso_plasma6`
+- `sd_image`
+- `sd_image_new_kernel`
+- `sd_image_new_kernel_no_zfs`
+
+This RFC offers absolutely no more guarantees than the current statu quo, as to whether proprietary packages will or not build on hydra.
+In particular, proprietary packages will not necessarily be part of the Zero Hydra Failures project upon release,
+though release managers could, at their own discretion, decide to include some specific proprietary packages in there.
+
+# Examples and Interactions
+[examples-and-interactions]: #examples-and-interactions
+
+With these changes, here is what could happen as things currently stand, if the licenses were all to be marked `runnableOnHydra`.
+This is not meant to be indicative of what should happen or not, but indicative of what could happen.
+Each package's individual `license` field setup is left to its maintainers, and nixpkgs governance should conflict arise.
+This RFC does not mean to indicate that it is right or wrong, and is not the right place to discuss changes to this field.
+Should one have disagreements on any specific package in this list, please bring that up to that package's maintainers.
+
+It is also suggested in this RFC that people, upon marking licenses as `runnableOnHydra`, check all the derivations that use this license.
+They could then have to mark them as either `hydraPlatforms = []`, `preferLocalBuild = true` and/or `allowSubstitutes = false`.
+This might be useful for packages like TPTP:
+they may not yet be marked as such due to these flags having no impact on unfree packages;
+but would take gigabytes on Hydra for basically no local build time improvement
+
+With this in mind, Hydra could start building, among others:
+- CUDA
+- DragonflyDB
+- MongoDB
+- Nomad
+- NVIDIA drivers
+- Outline
+- SurrealDB
+- TeamSpeak
+- Terraform
+- Unrar
+- Vagrant
+- NixOS tests that involve such software (eg. MongoDB or Nomad)
+
+And Hydra will keep not building, among others:
+- CompCert
+- DataBricks
+- Elasticsearch
+- GeoGebra
+- Widevine CDM
```

Comment:
Maybe it’d be best to remove the list of examples entirely if this isn’t meant to spark discussion on whether the “could build”/“wouldn’t build” line drawn here is accurate?